x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -400,14 +425,16 @@ private ProfileDocument updateWithSubject(ProfileDocument doc, Subject subject)
     // Package private for testing
     record VersionedDocument(ProfileDocument doc, long primaryTerm, long seqNo) {
 
-        Profile toProfile(@Nullable String realmDomain) {
-            return toProfile(realmDomain, Set.of());
-        }
-
-        Profile toProfile(@Nullable String realmDomain, @Nullable Set<String> dataKeys) {
+        /**
+         * Convert the index document to the user-facing Profile by
+         * (1) filtering through the application data
+         * (2) replace the domain with the most appropriate one. This means the latest authentication object
+         *     if one exists or extraction from the document itself.
+         */
+        Profile toProfile(Set<String> dataKeys, @Nullable RealmDomain realmDomain) {
```

Comment:
Same as above, I believe we don't need to supply the domain here, no?

Reply:
See above.